Integrate the PAKE Extension into the Crypto API specification (#177)
ARM-software/psa-api: doc/crypto/overview/functionality.rst

    designed to work only with short-term keys, or only with long-term
    non-extractable keys, do not need to provide this function.

Cryptographic operations
------------------------

For each type of cryptographic operation, the API can include:

> *   One or more *single-part* functions, that carry out one function of that operation. For example, compute, verify, encrypt, or decrypt.

Reviewer: Nit / editorial: I found this wording slightly confusing. Would something like:

```suggestion
*   One or more *single-part* functions, which perform a whole operation in a single function call. For example, compute, verify, encrypt or decrypt.
```
Work as an alternative?

.. _multi-part-operations:

Multi-part operations
~~~~~~~~~~~~~~~~~~~~~

Multi-part operations are APIs which split a single cryptographic operation into

> a sequence of separate steps. This enables fine control over the configuration

Author: I've updated the text along these lines, but keeping consistency with the wording of the setup functions.

I also took the opportunity to add PAKE as an example of why multi-part operations are needed.
